package/upgrade/lib.sh

```diff
@@ -8,10 +8,24 @@ UPGRADE_REPO_BUNDLE_METADATA="$UPGRADE_REPO_URL/bundle/metadata.yaml"
 CACHED_BUNDLE_METADATA=""
 HOST_DIR="${HOST_DIR:-/host}"
 
+download_file()
+{
+  local url=$1
+  local output=$2
+
```

Comment:
Let's add a statement like "Downloading xxx to yyy...".
Do we want to use the `--retry` flag of curl?

Reply:
Updated.
The flag does not support something like `-1` for endless retry, so we just go with the `until` loop.

Reply:
Updated from endless retry to limited retry since we're gonna have job-level retry.